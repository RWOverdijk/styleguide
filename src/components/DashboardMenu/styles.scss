/** @define DashboardMenu */

@import '../../css/variables';

.Van-DashboardMenu {
  position: relative;
  height: 50px;
  background-color: $color-white;
  border-top: 1px solid $color-grey-bright;
  border-bottom: 1px solid $color-grey-bright;

  &-nav {
    margin: 0;
    display: block;
    height: 50px;
    list-style: none;
    white-space: nowrap;
    padding: 0 15px;
  }

  &-item {
    display: inline-block;
    margin-right: 45px;
    font-size: $font-size-normal;
    font-weight: $font-weight-semibold;
    color: $color-dark;
    height: 100%;
    line-height: 50px;
    cursor: pointer;

    /* postcss-bem-linter: ignore */
    > a {
      color: $color-dark;
      text-decoration: none;
      display: block;

      /* postcss-bem-linter: ignore */
      &:hover,
      &:active,
      &:focus {
        color: $color-green-bright;
        text-decoration: none;
      }
    }

    &--active {
      /* postcss-bem-linter: ignore */
      &,
      > a {
        color: $color-green-bright !important;
      }
    }
  }

  @media (max-width: $container-sm) {
    overflow-y: hidden;
    overflow-x: scroll;
    -webkit-overflow-scrolling: touch;
    -ms-overflow-style: -ms-autohiding-scrollbar;
  }
}
