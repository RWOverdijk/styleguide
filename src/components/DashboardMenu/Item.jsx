import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './styles.scss';

const Item = ({ active, className, onClick, children, ...props }) => (
  <li
    className={classnames(
      'Van-DashboardMenu-item',
      { 'Van-DashboardMenu-item--active': active },
      className
    )}
    role="presentation"
    onClick={onClick}
    {...props}
  >
    {children}
  </li>
);

Item.defaultProps = {
  children: null,
  active: false,
  onClick: undefined,
  className: ''
};

Item.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  onClick: PropTypes.func,
  active: PropTypes.bool
};

export default Item;
