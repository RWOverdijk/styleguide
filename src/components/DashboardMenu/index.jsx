import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Item from './Item';
import './styles.scss';

class DashboardMenu extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      isStartShadowVisible: false,
      isEndShadowVisible: false
    };
  }

  componentDidMount() {
    if (this.navRef instanceof HTMLElement) {
      this.handleShadowVisibility(this.navRef);
    }
  }

  setRef(elem) {
    this.navRef = elem;
  }

  handleShadowVisibility = target => {
    const maxScrollStartReached = target.scrollLeft <= 1;
    const maxScrollEndReached = target.scrollLeft >= target.scrollWidth - target.offsetWidth;

    this.setState({
      isStartShadowVisible: !maxScrollStartReached,
      isEndShadowVisible: !maxScrollEndReached
    });
  };

  render() {
    const { isStartShadowVisible, isEndShadowVisible } = this.state;
    const { className, children, ...props } = this.props;

    return (
      <nav
        className={classnames('Van-DashboardMenu', className, {
          'Van-DashboardMenu--startShadow': isStartShadowVisible,
          'Van-DashboardMenu--endShadow': isEndShadowVisible
        })}
        {...props}
      >
        <ul
          className="Van-DashboardMenu-nav"
          onScroll={e => this.handleShadowVisibility(e.target)}
          ref={elem => this.setRef(elem)}
        >
          {children}
        </ul>
      </nav>
    );
  }
}

DashboardMenu.defaultProps = {
  children: null,
  className: null
};

DashboardMenu.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
};

DashboardMenu.Item = Item;

export default DashboardMenu;
