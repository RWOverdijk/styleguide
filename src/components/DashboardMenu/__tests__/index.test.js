import React from 'react';
import renderer from 'react-test-renderer';
import DashboardMenu from '../';

describe('<DashboardMenu>', () => {
  it("renders DashboardMenu", () => {
    const tree = renderer.create(
      <DashboardMenu>
        <DashboardMenu.Item>
          Overzicht
        </DashboardMenu.Item>
        <DashboardMenu.Item>
          Laadpassen
        </DashboardMenu.Item>
      </DashboardMenu>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders active item", () => {
    const tree = renderer.create(
      <DashboardMenu>
        <DashboardMenu.Item active>
          Overzicht
        </DashboardMenu.Item>
        <DashboardMenu.Item>
          Laadpassen
        </DashboardMenu.Item>
      </DashboardMenu>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders menu with a link element", () => {
    const tree = renderer.create(
      <DashboardMenu>
        <DashboardMenu.Item>
          <a href="#">Overzicht</a>
        </DashboardMenu.Item>
        <DashboardMenu.Item>
          <a href="#">Laadpassen</a>
        </DashboardMenu.Item>
      </DashboardMenu>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <DashboardMenu className="my-class">
        <DashboardMenu.Item className="my-class">
          Overzicht
        </DashboardMenu.Item>
        <DashboardMenu.Item>
          Laadpassen
        </DashboardMenu.Item>
      </DashboardMenu>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <DashboardMenu data-id="0">
        <DashboardMenu.Item data-id="0">
          Overzicht
        </DashboardMenu.Item>
        <DashboardMenu.Item>
          Laadpassen
        </DashboardMenu.Item>
      </DashboardMenu>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
