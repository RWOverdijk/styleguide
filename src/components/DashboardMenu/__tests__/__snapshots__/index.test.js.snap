// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardMenu> renders DashboardMenu 1`] = `
<nav
  className="Van-DashboardMenu"
>
  <ul
    className="Van-DashboardMenu-nav"
  >
    <li
      className="Van-DashboardMenu-item"
      onClick={undefined}
      role="presentation"
    >
      Overzicht
    </li>
    <li
      className="Van-DashboardMenu-item"
      onClick={undefined}
      role="presentation"
    >
      Laadpassen
    </li>
  </ul>
</nav>
`;

exports[`<DashboardMenu> renders active item 1`] = `
<nav
  className="Van-DashboardMenu"
>
  <ul
    className="Van-DashboardMenu-nav"
  >
    <li
      className="Van-DashboardMenu-item Van-DashboardMenu-item--active"
      onClick={undefined}
      role="presentation"
    >
      Overzicht
    </li>
    <li
      className="Van-DashboardMenu-item"
      onClick={undefined}
      role="presentation"
    >
      Laadpassen
    </li>
  </ul>
</nav>
`;

exports[`<DashboardMenu> renders menu with a link element 1`] = `
<nav
  className="Van-DashboardMenu"
>
  <ul
    className="Van-DashboardMenu-nav"
  >
    <li
      className="Van-DashboardMenu-item"
      onClick={undefined}
      role="presentation"
    >
      <a
        href="#"
      >
        Overzicht
      </a>
    </li>
    <li
      className="Van-DashboardMenu-item"
      onClick={undefined}
      role="presentation"
    >
      <a
        href="#"
      >
        Laadpassen
      </a>
    </li>
  </ul>
</nav>
`;

exports[`<DashboardMenu> renders with specific attributes 1`] = `
<nav
  className="Van-DashboardMenu"
  data-id="0"
>
  <ul
    className="Van-DashboardMenu-nav"
  >
    <li
      className="Van-DashboardMenu-item"
      data-id="0"
      onClick={undefined}
      role="presentation"
    >
      Overzicht
    </li>
    <li
      className="Van-DashboardMenu-item"
      onClick={undefined}
      role="presentation"
    >
      Laadpassen
    </li>
  </ul>
</nav>
`;

exports[`<DashboardMenu> renders with specific className 1`] = `
<nav
  className="Van-DashboardMenu my-class"
>
  <ul
    className="Van-DashboardMenu-nav"
  >
    <li
      className="Van-DashboardMenu-item my-class"
      onClick={undefined}
      role="presentation"
    >
      Overzicht
    </li>
    <li
      className="Van-DashboardMenu-item"
      onClick={undefined}
      role="presentation"
    >
      Laadpassen
    </li>
  </ul>
</nav>
`;
